frontend/app/src/components/home/RightPanel.svelte

```diff
@@ -51,15 +52,10 @@
     $: groupChat = selectedChatStore as Readable<GroupChatSummary>;
     $: empty = $rightPanelHistory.length === 0;
 
-    function onDismissAsAdmin(ev: CustomEvent<string>): void {
+    function onChangeRole(ev: CustomEvent<{ userId: string; newRole: MemberRole; oldRole: MemberRole; promotion: boolean }>): void {
```

Comment:
This causes a slight wince as `promotion` is a function of `newRole` and `oldRole` so passing them around separately seems a bit woolly.